packages/@lwc/engine-core/src/framework/vnodes.ts

```diff
@@ -144,3 +145,12 @@ export function isVFragment(vnode: VNode): vnode is VFragment {
 export function isVScopedSlotFragment(vnode: VNode): vnode is VScopedSlotFragment {
     return vnode.type === VNodeType.ScopedSlotFragment;
 }
+
+export function areDynamicVCustomElements(vnode1: VNode, vnode2: VNode): boolean {
```

Comment:
Nit: This function would be more reusable if it were `isDynamicVCustomElement`, with only one required param. Where it is currently consumed, that would mean `isDynamicVCustomElement(vn1) && isDynamicVCustomElement(vn2)` in place of `areDynamicVCustomElements(vn1, vn2)`. Both implementations would be inlined by the JS engine, so this change shouldn't introduce a performance hit.

Reply:
Thanks for the feedback Dale!  I ended up [simplifying](https://github.com/salesforce/lwc/pull/3338/commits/b7dcd96d43735e82762883ae19b7b9057bce2479#diff-8622743d79e146365dd8c59fb85057a09b0f9577d99f065628abdd95d0d7a754R92-R100) the check and removed the helper function entirely.